src/js/shield_canvas_draw.js

```diff
@@ -341,39 +351,51 @@ export function homePlate(
 
   let lineThick = outlineWidth * PXR;
   let lineWidth = lineThick / 2;
-  let drawRadius = radius * PXR;
-  let drawOffset = Math.abs(offset) * PXR;
+  let drawRadius1 = radius1 * PXR;
+  let drawRadius2 = radius2 * PXR;
+  let drawOffset = offset * PXR;
 
   let x0 = lineWidth;
-  let x4 = width - lineWidth;
-  let y0 = lineWidth;
-  let y4 = CS - lineWidth;
+  let x12 = width - lineWidth;
+  let y0 = pointUp ? CS - lineWidth : lineWidth;
+  let y5 = pointUp ? lineWidth : CS - lineWidth;
 
-  let x1 = x0 + drawRadius;
-  let x2 = (x0 + x4) / 2;
-  let x3 = x4 - drawRadius;
-  let y1 = y0 + drawRadius;
-  let y3 = y4 - drawOffset;
+  let y3 = y5 - angleSign * drawOffset;
 
-  let drawOffsetTangent =
-    drawRadius * Math.tan(Math.PI / 4 - Math.asin(drawOffset / (x2 - x0)) / 2);
-  let y2 = y3 - drawOffsetTangent;
-
-  if (offset < 0) {
-    y0 = CS - y0;
-    y1 = CS - y1;
-    y2 = CS - y2;
-    y3 = CS - y3;
-    y4 = CS - y4;
-  }
+  let x4 = x0 + angleSign * (y3 - y0) * tangent;
+  let x6 = (x0 + x12) / 2;
+  let x8 = x12 - angleSign * (y3 - y0) * tangent;
+  // let y1 = y0 + radius2 * (1 - sine);
+
+  let offsetAngle = Math.atan(drawOffset / (x6 - x0));
+  let offsetSine = Math.sin(offsetAngle);
+  let offsetCosine = Math.cos(offsetAngle);
+
+  let halfComplementAngle1 = (Math.PI / 2 - offsetAngle + angle) / 2;
+  let halfComplementTangent1 = Math.tan(halfComplementAngle1);
+
+  let halfComplementAngle2 = (Math.PI / 2 - angle) / 2;
+  let halfComplementTangent2 = Math.tan(halfComplementAngle2);
+
+  let x1 = x0 + angleSign * drawRadius1 * halfComplementTangent1 * sine;
+  // let x2 = x0 + drawRadius1 * halfComplementTangent1 * offsetCosine;
+  let x5 = x4 + drawRadius2 * halfComplementTangent2;
+  let x7 = x8 - drawRadius2 * halfComplementTangent2;
+  // let x10 = x12 - drawRadius1 * halfComplementTangent1 * offsetCosine;
+  let x11 = x12 - angleSign * drawRadius1 * halfComplementTangent1 * sine;
+  let y2 = y3 - angleSign * drawRadius1 * halfComplementTangent1 * cosine;
+  // let y4 = y3 + angleSign * drawRadius1 * halfComplementTangent1 * offsetSine;
+
+  // let x3 = x5 - angleSign * drawRadius2 * sine;
+  // let x9 = x7 + angleSign * drawRadius2 * sine;
```

Comment:
Lots of commented code here?

Reply:
Commented lines of code removed; variable names decremented.